readme.md

```diff
@@ -17,89 +19,111 @@ In your `index.html` add a reference to our plugin.
 
 ```html
-    <script>
-        window.startRepairWise(getToken, saveReportInfo, address, redirectUrl);
-    </script>
+</body>
+<script>
+  // How to start the plugin and load it
+  var repairWise = window.RepairWise({
+    getToken: getToken,
+    onCloseUrl: redirectUrl,
+    onCloseRepairWise: onCloseRepairWise,
+    saveProjectInfo: saveProjectInfo,
+  })
+
+  if (projectId) {
+    repairWise.start({ startingProjectId: projectId })
+  } else if (address) {
+    repairWise.start({ startingAddress: address })
+  } else {
+    repairWise.start()
+  }
+</script>
 ```
 
-The plugin won't quite work yet, we need to implement the two required methods `getToken` and `saveReportInfo` in our `utils.js`.
+### Parameters
 
-First we'll add the `getToken` method.
+The `getToken` callback method is required for the plugin to work.
 
 ```javascript
   function getToken() {
     // callback to your OAuth resource
   }
 ```
+
 For more information on OAuth and getting a token read [OAuth](#oauth).
 
-Lastly we'll add the `saveReportInfo` in our `utils.js`.
+The `onCloseUrl` redirect is optional. Adding it will add a `Close` button in the top right that redirects the user to the provided url
+
+The `onCloseRepairWise` callback function will be called as repairwise is closing. It takes no parameters and returns nothing.
 
 ```javascript
-  function saveReportInfo(addressJson, linkToReport) 
-  { 
-     console.log(addressJson.propertyNumber); 
-     console.log(addressJson.street); 
-     console.log(addressJson.city); 
-     console.log(addressJson.stateProvince); 
-     console.log(addressJson.postalCode); 
-     console.log(addressJson.country); 
-    
-     console.log(linkToReport); 
-  } 
+function onCloseRepairWise()
+{
+  console.log("Closing RepairWise, please save any information");
+}
 ```
 
-We are just printing this information to the console, ultimately what you do with the `addressJson` and `linkToReport` is up to you.
+The `saveProjectInfo` callback is called when a new project is created in RepairWise.
+
+Note: This will happen right after the user puts in an address. If you started RepairWise with an address, the project is created as RepairWise starts up. If the address has invalid information, like a bad postal code, the user will be taken to the address entry screen and this callback will be called after the correct address is entered. If you start RepairWise with a projectId, this callback will not be called unless the the get for the project with that id fails for some reason. If the project with the id could not be loaded, the user will be taken to the address entry screen and this callback will be called after the address is entered and a new project is created. In other words, even in the case of starting with a potentially good project id, you should be prepared to re-store a new project id just in the case the one provided is bad.
```

Comment:
Minor typo
is:
"If you start RepairWise with a projectId, this callback will not be called unless the the get for the project with that id fails for some reason."
should be:
"If you start RepairWise with a projectId, this callback will not be called unless the get for the project with that id fails for some reason."

Reply:
Fixed